### PR TITLE
Feature/92 Provide Makefile

### LINKED ISSUE #92: Create Makefile
**Describe the solution you'd like**
Now we have docker-compose and we will have tests and so on, could be handy to have a nice Makefile to make things easier. For example with `make app` the app starts. With `make tests` all the tests run and so on.

**Describe alternatives you've considered**
I don't know if there are another tools, up to you.

### PR DESCRIPTION
**Pull request by XaviTorello**
## TL;DR
With this setup, we just create an specific layer with testing requirements also applied ready to work tagged locally in `anem-per-feina:tests`. Also it provides a `make` with most common development actions integrated

Fix #92 Create Makefile

## Description

It provides a `Makefile` devised to simplify development actions:

```make
$ make help
usage: make [target]

options:
  start                           Start all or c=<name> containers in FOREGROUND
  serve                           Start all or c=<name> containers in BACKGROUND
  test                            Execute tests
  stop                            Stop all or c=<name> containers
  restart                         Restart all or c=<name> containers
  status                          Show status of containers
  ps                              Alias of status
  clean                           Clean all containers (keeping volumes)
  purge                           Purge all containers and volumes
  build                           (re)Build all images or c=<name> containers

other:
  help                            Show this help.
  mk-upgrade                      Check for updates of mk-lib
  mk-version                      Show current version of mk-lib
```

The idea is that now our composition will build two local images:
- `anem-per-feina:local`: with current Dockerfile definition (just `requirements.txt` file)
- `anem-per-feina:tests`: extending `local` with testing requirements (processing `requirements-test.txt`)

This is handled with a new service named `test`:
- devised just to handle local testing image creation (on `build` or `up`)
- has no `entrypoint`, so it just dies when composition starts
- it's used to dispatch temporal runs of our testing scenario via `make test`
  - it patchs `test` service `entrypoint` to invoke our test task and dies (will be deleted once ends with `--rm`)

## Considerations

One option was to also process `requirements-test.txt` inside the base image, but this increases +53% image size:
```
$ docker images anem-per-feina
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
anem-per-feina      tests               d27527209a7c        14 minutes ago      317MB
anem-per-feina      local               55cd78c97c96        15 minutes ago      207MB
```

Remember that this base Dockerfile will be the one used for everything (CI, CD, ...)

Another option was to integrate the `requirements-test.txt` processing inside the `app` entrypoint, or inside an specific task to be invoked, but it's very slow and complicates the development experience.

We could also duplicate the docker-file with an specific testing definition, but maybe can introduce some non needed complexity. This way can be studied in short if we need it :) 

Used https://github.com/krom/docker-compose-makefile to simplify Makefile definition (just 40kB)

## Code formating

Done

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+.PHONY: help start serve test stop restart status ps clean purge build
+
+start: ## Start all or c=<name> containers in FOREGROUND
+	@$(DOCKER_COMPOSE) -f $(DOCKER_COMPOSE_FILE) up $(c)
+
+serve: ## Start all or c=<name> containers in BACKGROUND
+	@$(DOCKER_COMPOSE) -f $(DOCKER_COMPOSE_FILE) up $(c) -d
+
+test: ## Execute tests
```
Comment (jbagot) [inline Makefile:12]: Would be nice to have another option to execute an specific test or all tests from one folder as you can do it in pytest. What do you think?
I've seen that you used the django command to run the tests, but it will trigger the pytests? Can you add a parameter to add the specific test path or folder path?

File: .mk-lib/README.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
+# Docker compose makefile
```
Comment (jbagot) [inline .mk-lib/README.md:1]: I didn't know about this package, I'll check it later, but it adds a lot of files I don't know what is the added value

File: Makefile (hunk 1 above)
Comment (XaviTorello) [reply Makefile:12]: Sure, this is just an starting point :)

Tomorrow I'll explore options to inject strings or define specific string modifiers (filters).

Regarding the executed tests, it's not hardcoded, it's configurable on the invoked task so can be ammended without effort.

Currently there are no tests, pytest has not been configured and maybe it's better to deal with this in another PR

File: .mk-lib/README.md (hunk 2 above)
Comment (XaviTorello) [reply .mk-lib/README.md:1]: It speeds up targets generation, provides auto help and more enhancements.

Btw, this is not mandatory :)

**Comment by XaviTorello** [comment]
The proposed solution makes sense @jbagot? Or you want to explore another solution?

I would liked to reach more feedback about the abstract solution (not about the detaila about the tests params) :)

**Review by jbagot — APPROVED** [review]
It's fine, thanks @XaviTorello
